orderbook/src/database/orders.rs

```diff
@@ -235,11 +235,19 @@ impl OrderStoring for Postgres {
                 COALESCE(SUM(t.buy_amount), 0) AS sum_buy, \
                 COALESCE(SUM(t.sell_amount), 0) AS sum_sell, \
                 COALESCE(SUM(t.fee_amount), 0) AS sum_fee, \
-                (COUNT(invalidations.*) > 0 OR o.cancellation_timestamp IS NOT NULL) AS invalidated \
+                (COUNT(invalidations.*) > 0 OR o.cancellation_timestamp IS NOT NULL) AS invalidated, \
+                (o.signing_scheme = 'presign' AND COUNT(ps.*) = 0) AS presignature_pending \
             FROM \
                 orders o \
                 LEFT OUTER JOIN trades t ON o.uid = t.order_uid \
                 LEFT OUTER JOIN invalidations ON o.uid = invalidations.order_uid \
+                LEFT OUTER JOIN LATERAL ( \
+                    SELECT p.signed \
+                    FROM presignature_events as p \
+                    WHERE o.uid = p.order_uid \
+                    ORDER BY p.block_number DESC, p.log_index DESC \
+                    FETCH FIRST 1 ROW ONLY \
+                ) ps ON ps.signed \
```

Comment:
Because we only select one row from the presignature table a more natural way to write this is to use a sub query instead of joining:

```SQL
 (o.signing_scheme = 'presign' AND (
    SELECT p.signed
    FROM presignature_events p
    WHERE o.uid = p.order_uid
    ORDER BY p.block_number DESC, p.log_index DESC
    LIMIT 1
)) AS presignature_pending
```

If postgres complains that the subquery doesn't just return one row (not sure it is smart enough to understand LIMIT 1) then might have to transform it so that it uses an aggregate

```SQL
 (o.signing_scheme = 'presign' AND (
    SELECT COUNT(*)
    FROM presignature_events p
    WHERE o.uid = p.order_uid AND p.signed
    ORDER BY p.block_number DESC, p.log_index DESC
    LIMIT 1
)) AS presignature_pending
```

Reply:
Sounds good. I was under the impression that a lateral join ended up doing an equivalent thing, but I may be wrong.

Reply:
I think they are equivalent in what we observe but I feel the sub query is a more natural way to write it. When I tested some other JOIN LATERAL before I also saw that they weren't optimized as well.

Reply:
Ended up having to go with something like:
```sql
    COALESCE((o.signing_scheme = 'presign' AND (
        SELECT (NOT p.signed) as unsigned
        FROM presignature_events p
        WHERE o.uid = p.order_uid
        ORDER BY p.block_number DESC, p.log_index DESC
        LIMIT 1
    )), true) AS presignature_pending
```
The `COALESCE` is needed to deal with the case of no presignatures.

Reply:
Also worth noting that the `most_recent_with_orderuid` index is used :tada:.